### PR TITLE
feat: SQS user can now trigger DLQ redrive

### PR DESCRIPTION
**Pull request by FelisiaM**
The user created as a result of a bind operation for the (original) source queue has permissions to trigger a redrive operations for messages to go from DLQ to original queue

### Checklist:

* [ ] Have you added Release Notes in the docs repositories?
* [x] Have you ran `make run-integration-tests` and `make run-terraform-tests`?
* [x] Have you ran acceptance tests for the service under change?
* [x] Have you followed the [Conventional Commits specification](https://www.conventionalcommits.org/en/v1.0.0/#summary)?

### CODE REVIEW

File: acceptance-tests/sqs_test.go [hunk 1]
```diff
@@ -34,21 +34,22 @@ var _ = Describe("SQS", Label("sqs"), func() {
 		By("checking that the app environment has a credhub reference for credentials")
 		Expect(binding.Credential()).To(HaveKey("credhub-ref"))
 
-		By("sending a message using the first app")
+		By("sending a message from producer app")
```
Comment (blgm) [inline acceptance-tests/sqs_test.go:37]: I much prefer this terminology

File: acceptance-tests/apps/sqsapp/internal/app/redrive.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+		cred, ok := creds[binding]
+		if !ok {
+			return http.StatusBadRequest, fmt.Sprintf("no creds found for binding: %q", binding)
+		}
+		cfg, err := cred.Config()
+		if err != nil {
+			return http.StatusInternalServerError, fmt.Sprintf("could not read AWS config: %q", err)
+		}
+
+		dlqBinding := r.URL.Query().Get("dlq_binding")
```
Comment (blgm) [inline acceptance-tests/apps/sqsapp/internal/app/redrive.go:29]: There's something a bit messy about taking one binding name as a URL path and another as a URL parameter, but I can't think of a better way, and this is just a test app, not a production-grade REST API...

**Comment by blgm** [comment]
This is great. One thought that I did have is that we seem to be building the policy with:
```
data "aws_iam_policy_document" "user_policy" {
  dynamic "statement" {
    for_each = local.queue_policy
    content {
      sid       = try(statement.value.sid, null)
      actions   = try(statement.value.actions, null)
      resources = try(statement.value.resources, null)
    }
  }
}
```
And I wonder whether this is still needed now that we don't have a policy switch. We used to define the policy [like this](https://github.com/cloudfoundry/csb-brokerpak-aws/blob/d784b374d1a6c3660b0f1957e365a187cf59c632/terraform/sqs/bind/data.tf#L1C1-L15C2), which avoids the `dynamic` keyword:
```
data "aws_iam_policy_document" "user_policy" {
  statement {
    sid = "sqsAccess"
    actions = [
      "sqs:SendMessage",
      "sqs:ReceiveMessage",
      "sqs:DeleteMessage",
      "sqs:PurgeQueue",
      "sqs:GetQueueAttributes"
    ]
    resources = [
      var.arn
    ]
  }
}
```

**Review by blgm — APPROVED** [review]

File: acceptance-tests/apps/sqsapp/internal/app/redrive.go (hunk 2 above)
Comment (FelisiaM) [reply acceptance-tests/apps/sqsapp/internal/app/redrive.go:29]: Definitely agree its ugly API design. Thinking out it we just need the ARN of the DLQ as a param which we actually have in the test. I will make a change to pass the DLQ ARN.

**Pull request merged**